BZ 1259118: Added file on available ports

bfallonf opened the pull request:
@adellape @tnguyen-rh @tpoitras @CowboysFan I'm not sure if this is in need of a tech review, as it's pretty much just the KBase article, but this is to do with this BZ:

https://bugzilla.redhat.com/show_bug.cgi?id=1259118

As discussed with Bilhar, it might be better to focus on an actual Troubleshooting Guide of sorts after the 3.1 release (and his impending restructure), but an available ports doc is something people have been asking for ( @thoraxe I'm looking at you). 

This is open to suggestion (in case I've done something off), but it's the info from:
- this KBase: https://access.redhat.com/solutions/1520653
- this email thread (internal): http://post-office.corp.redhat.com/archives/openshift-sme/2015-June/msg00543.html
- and made from this GH issue: #684 

Any thoughts?

tnguyen-rh commented:
@bfallonf 
Two suggestions:

- How about using tables?  (style)
- Sort (numeric, ascending) the entries in each group.  (discoverability)

Otherwise, LGTM.

adellape commented:
> How about using tables? (style)
> Sort (numeric, ascending) the entries in each group. (discoverability)

+1 to both

@bfallonf Typo in the file name (`avaialble_ports.adoc`) is what caused Travis to fail.

adellape commented:
@bfallonf FYI, https://bugzilla.redhat.com/show_bug.cgi?id=1270960 is also about creating a port map doc like this.

bfallonf commented:
@tnguyen-rh @adellape I had it as tables originally, but didn't like the 'one item' tables. Changed it back tho. I also added the IaaS stuff from the BZ Alex linked in.

Any more thoughts?

tnguyen-rh commented:
@bfallonf LGTM

adellape commented on admin_guide/available_ports.adoc:
```diff
@@ -0,0 +1,117 @@
+
+== Overview
+
+OpenShift infrastructure components talk to each other using ports;
```
"Talk" seems like an anthropomorphism ([Style Guide](https://obriend.fedorapeople.org/WritingStyleGuide/#sect-Red_Hat_Technical_Publications-Writing_Style_Guide-Avoiding_Slang_Metaphors_and_Misleading_Language-Anthropomorphism)). "communicate with each other" instead?

adellape commented on admin_guide/available_ports.adoc:
```diff
@@ -0,0 +1,117 @@
+== Overview
+
+OpenShift infrastructure components talk to each other using ports;
+communication endpoints that are identifiable for specific processes or
```
Maybe I'm just not familiar w/ this type of semi-colon usage, but shouldn't the part after the semi-colon be a complete sentence?

adellape commented on admin_guide/available_ports.adoc:
```diff
@@ -0,0 +1,117 @@
+|Required for SDN communication between pods on separate hosts.
+
+| *443* or *8443*
+|Required for the nodes to communicate to the master API, for the nodes to post back status, to receive tasks, et al..
```
Seems like this should just be one period on the end, even if "et al." usually has its own. Mild Googling appears to agree, but maybe I'm wrong.

bfallonf replied:
On second thoughts, IBM style guide says not to use latin abbreviations. Changed to "and so on", which it suggests.

adellape replied:
:+1:

adellape commented on admin_guide/available_ports.adoc:
```diff
@@ -0,0 +1,117 @@
+|Required for SDN communication between pods on separate hosts.
+
+| *10250*
+|On Kubelet minions, the master proxies to nodes via the Kubelet for `oc`
```
I take it this is a translation of "on the minons(Kublet)" from the KBase? Minions got renamed to nodes a while ago, so that should be the usage here, but I think you can safely just drop everything before the period instead.

adellape commented on admin_guide/available_ports.adoc:
```diff
@@ -0,0 +1,117 @@
+|Must be the port on the master host. Provides DNS services.
+
+| *4001*
+|Must be the port on the master host. Used for etcd to accept changes in state.
```
The "Must be the port on the master host" in the above two entries seem strange. Could probably just drop them.

adellape commented on admin_guide/available_ports.adoc:
```diff
@@ -0,0 +1,117 @@
+[cols='2,8']
+|===
+| *443* or *8443*
+|Required for the nodes to communicate to the master API, for the nodes to post back status, to receive tasks, et al..
```
Same wrt "et al."

adellape commented on admin_guide/available_ports.adoc:
```diff
@@ -0,0 +1,117 @@
+* In the above examples, port *4789* is used for User Datagram Protocol (UDP).
+* When deployments are using the SDN, the pod network is accessed via a service proxy, unless it is accessing the registry from the same node the registry is deployed on.
+* OpenShift internal DNS cannot be received over SDN. Depending on the detected values of `*openshift_facts*`, or if the `*openshift_ip*` and `*openshift_public_ip*` values are overridden, it will be the computed value of `*openshift_ip*`. For non-cloud deployments, this will default to the IP address associated with the default route on the master host. For cloud deployments, it will default to  ...
+* The master host uses port *10250* to reach the nodes and doesn't go over SDN. It depends on the target host of the deployment and uses the computed values of `*openshift_hostname*` and `*openshift_public_hostname*`.
```
s/doesn't/does not/

adellape commented:
@bfallonf Comments inline. There's also a number of lines that can be wrapped at 80.

Also the build_cfg is likely what's causing the current conflict cuz another new topic was merged today (the LDAP one).

bfallonf commented:
@adellape @tnguyen-rh Thanks for the thumbs up, comments. Alex, I addressed your comments above, and responded to one. If nothing else, this should be :ok: .

tnguyen-rh commented:
@adellape 
Is there a wrapping policy or guideline?
This is one of the first things i wondered about but never found the opportunity to dig into.

adellape commented:
> Is there a wrapping policy or guideline?

@tnguyen-rh Yeah, hard wrap at 80 column lines to make it easier to read the source consistently. It doesn't look like that ever made it into https://github.com/openshift/openshift-docs/blob/master/contributing_to_docs/doc_guidelines.adoc, but it needs to be.

adellape commented:
@bfallonf :+1:

adellape commented:
@bfallonf How would you feel about a rename of the title to either "Port Availability" or "Port Requirements" (personally prefer the latter)? "Available Ports" in a vacuum sounds like "these are some ports that are available", whereas what we're really saying is "these are ports that OpenShift needs to communicate on and requires to be open."

If you're into it, I can do so at the same time in a PR I'm planning which moves some early config topics (like this one, auth config, etc) from the Admin Guide up into the Install/Config guide.

bfallonf commented:
@adellape I'm fine with that. I was questioning the title anyway. Go ahead :fire_engine:

adellape commented:
I've removed the milestone from this and am not adding a branch label because there was follow-up work done to the port map tables, including fixes and moving it to the Prereqs topic. All that stuff is in the 3.1 docs already, but it still needs to get into the 3.0 docs per the related BZ (https://bugzilla.redhat.com/show_bug.cgi?id=1270960). At this point I'd rather just do the work directly on the enterprise-3.0 branch versus trying to cherrypick this, considering the time that's passed.

I'll continue to track in the BZ.
